#!/bin/sh
source="$(find /usr/lib/python*.*/site-packages/configgen/generators/citra -name citraGenerator.py)"
cp -a "${source}" "${source}.backup-language"
patch "${source}" << 'EOF'
--- ${source}
+++ citraGenerator.py
@@ -225,11 +225,26 @@
             return "left"
         return "unknown"

-# Lauguage auto setting
+# Language auto setting
 def getCitraLangFromEnvironment():
+    region = { "AUTO": -1, "JPN": 0, "USA": 1, "EUR": 2, "AUS": 3, "CHN": 4, "KOR": 5, "TWN": 6 }
+    availableLanguages = {
+        "ja_JP": "JPN",
+        "en_US": "USA",
+        "de_DE": "EUR",
+        "es_ES": "EUR",
+        "fr_FR": "EUR",
+        "it_IT": "EUR",
+        "hu_HU": "EUR",
+        "pt_PT": "EUR",
+        "ru_RU": "EUR",
+        "en_AU": "AUS",
+        "zh_CN": "CHN",
+        "ko_KR": "KOR",
+        "zh_TW": "TWN"
+    }
     lang = environ['LANG'][:5]
-    availableLanguages = { "ja_JP": 0, "en_US": 1, "fr_FR": 2, "de_DE": 3, "it_IT": 4, "es_ES": 5, "zh_CN": 6, "ko_KR": 7, "hu_HU": 8, "pt_PT": 9, "ru_RU": 10, "zh_TW": 11 }
     if lang in availableLanguages:
-        return availableLanguages[lang]
+        return region[availableLanguages[lang]]
     else:
-        return availableLanguages["en_US"]
+        return region["AUTO"]
EOF

echo >&2 "Citra Generator was patched for language fix."
